### PR TITLE
Fix Algorithm.trainable_variables

### PR DESCRIPTION
**Pull request by emailweixu**

### CODE REVIEW

File: alf/algorithms/algorithm.py [hunk 1]
```diff
@@ -157,8 +157,9 @@ def _is_alg(obj):
             return isinstance(obj, Algorithm)
 
         def _is_module_or_var(obj):
-            return (isinstance(obj, (tf.Variable, tf.Module))
-                    and not isinstance(obj, Algorithm))
+            return ((isinstance(obj, tf.Variable) and obj.trainable)
```
Comment (hnyu) [inline alf/algorithms/algorithm.py]: if obj.trainable is False, then obj.trainable_variables will be [], which is fine in that case

Comment (emailweixu) [reply alf/algorithms/algorithm.py]: Variable does not have property trainable_variables

Comment (Haichao-Zhang) [reply alf/algorithms/algorithm.py]: Should the function name be updated to something like "_is_module_or_trainable_var" in order to reflect its functionality?

Comment (emailweixu) [reply alf/algorithms/algorithm.py]: It's fine as it's a local function which only used here.

**Pull request merged**